### DISCUSSION TITLE
Assigning IP to devices and interfaces in v3.5.1

### DISCUSSION THREAD

**Discussion by tfboy** [netbox-community/netbox#12577] [Help Wanted!]
Having just upgraded to v3.5.1, I've just noticed that the assignment section has been modified and now only shows the Interfaces entry box. In previous versions, there was first a Device (or Virtual Machine) box which would filter based on device to then list only the interfaces present on that device / VM.

Now, that first level filter is gone.

It looks like I can bring up a filter selection tool by clicking on the icon to the right, but it seems strange to have to do this. But with the current method of operation, this seems mandatory: I have like 100 "Gigabit Ethernet 0/1" interfaces with no indication which device each one belongs so it's totally useless having just the "Interface" box to select from.

Is this by design and I'm missing a trick or is this a bug?

**Comment by a084ed22** [comment]
The new interface allows to define additional filters compared to the previous system and to apply filter on different properties rather than only the device name.

**Reply by tfboy** [reply]
Yes, but it still seems a strange way of doing things.
Having just "Interfaces" on its own is pretty useless in my opinion. In wihch case the additional filters button is always going to be used. Why show a useless field as you have to do multiple mouse clicks to bring up a filter selection, add device (or whatever else).

I like the idea of having the new filter button, that's really useful, but would seriously request the original "device" box is restored so people don't have to change the long-standing way of doing things if it was perfectly functional.

**Reply by carlo-4ip** [reply]
I second this, I also am missing the old device field.

The new search box is a great addition, but for quickly creating new IP addresses it's really handy to have a selection based on:
- the tenant, 
- then a device field (which only shows devices of the selected tenant) 
- and last the interface field (that only shows the interfaces of the selected device). 

99 out of 100 times we connect an IP to an interface based on these 3 values.

**Reply by tfboy** [reply]
Having now had to update a few dozen IPs, in my opinion, the removal of the "device" search box is just mind-numbingly infuriating. I agree having the extra filter button is a great addition but please bring back the "device" and "interface" fields natively to the IP address page. Every time I want to change something, I have to click on the filter, then select device in the menu on the left then type the device name, then click elsewhere, then click search to then have the filtered interface list. It's so time-consuming! :(

**Comment by carlo-4ip** [comment]
Dear developers,

Can you please do something with this issue?

The assignment of IP adresses to devices is very difficult compared to how it was in earlier versions.

It would be very useful to have tenant and device fields in the "Interface Assignment" of an IP Address, which filters each following field, so there are limited interfaces left. 
The "Select Interface" search window that is there now is too complex for quick searches, but can still exist as extra filter for less common searches.
